activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/scheduler/JobSchedulerImpl.java

```diff
@@ -797,6 +804,12 @@ protected void mainLoop() {
                 } finally {
                     this.store.readUnlockIndex();
 
+                    // deferred execution of all jobs to be scheduled to avoid deadlock with indexLock
+                    for (Runnable command : toSchedule) {
+                        command.run();
```

Comment:
I think we may need some sort of catch-and-continue pattern here, since if scheduling one fails due to IOException, the remaining would all fail to _try_ to schedule.